### PR TITLE
GDB-11059 Improve The Logging Of Kafka Sink

### PR DESCRIPTION
**Pull request by Secchol**
Added more detailed logs in all of the Kafka-Sink classes to make debugging easier and the record consumption process more understandable.

### CODE REVIEW

File: src/main/java/com/ontotext/kafka/service/SinkRecordsProcessor.java [hunk 1]
```diff
@@ -132,16 +144,28 @@ public Object call() throws RetriableException {
 		long start = System.currentTimeMillis();
 		try (RepositoryConnection connection = repository.getConnection()) {
 			connection.begin();
+			int recordsInCurrentBatch = recordsBatch.size();
+			if (LOG.isTraceEnabled()) {
```
Comment (ivorusev) [inline src/main/java/com/ontotext/kafka/service/SinkRecordsProcessor.java:148]: Why `LOG.isTraceEnabled()`, it makes sense to use it when you have additional operations that are related to logging the trace info. But for simple logging makes no sense.

File: src/main/java/com/ontotext/kafka/service/SinkRecordsProcessor.java [hunk 2]
```diff
@@ -132,16 +144,28 @@ public Object call() throws RetriableException {
...
+						this.repositoryUrl, batchSize, recordsInCurrentBatch);
+			}
 			while (!recordsBatch.isEmpty()) {
 				SinkRecord record = recordsBatch.remove();
 				if (!failedRecords.contains(record)) {
 					handleRecord(record, connection);
 				}
 			}
 			connection.commit();
+			if (LOG.isTraceEnabled()) {
```
Comment (ivorusev) [inline src/main/java/com/ontotext/kafka/service/SinkRecordsProcessor.java:160]: Why LOG.isTraceEnabled(), it makes sense to use it when you have additional operations that are related to logging the trace info. But for simple logging makes no sense.

File: src/main/java/com/ontotext/kafka/service/UpdateRecordsProcessor.java [hunk 3]
```diff
@@ -34,14 +39,23 @@ protected UpdateRecordsProcessor(Queue<Collection<SinkRecord>> sinkRecords, Atom
...
-				.execute();
+					.execute();
 			connection.add(ValueUtil.convertRDFData(record.value()), format);
+			long finish = System.currentTimeMillis();
+			if (LOG.isTraceEnabled()) {
+				LOG.trace("Record info: {}", ValueUtil.recordInfo(record));
+				LOG.trace("Converted the record and added it to the RDF4J connection for {} ms", finish - start);
+			}
 		} catch (IOException e) {
+			LOG.debug("Caught an I/O exception while processing record");
```
Comment (ivorusev) [inline src/main/java/com/ontotext/kafka/service/UpdateRecordsProcessor.java:54]: Not sure if those should be in debug. wdyt?

File: src/main/java/com/ontotext/kafka/util/PropertiesUtil.java [hunk 4]
```diff
@@ -27,7 +27,7 @@
```
Comment (ivorusev) [inline src/main/java/com/ontotext/kafka/util/PropertiesUtil.java:27]: I don't see the changes where we log the excepliton when validating the /versions url. Did you skip those on purpose?

File: src/main/java/com/ontotext/kafka/service/GraphDBService.java [hunk 5]
```diff
@@ -79,11 +81,17 @@ private Repository initializeRepository(Repository repository, Map<String, ?> pr
...
 		String address = (String) properties.get(GraphDBSinkConfig.SERVER_IRI);
 		String repositoryId = (String) properties.get(GraphDBSinkConfig.REPOSITORY);
+		LOG.trace("Fetching repository from {}", address);
+		LOG.trace("Repository name: {}", repositoryId);
 		var repository = new HTTPRepository(address, repositoryId);
 		switch (GraphDBSinkConfig.AuthenticationType.of((String) properties.get(GraphDBSinkConfig.AUTH_TYPE))) {
 			case NONE:
 				return repository;
 			case BASIC:
+				if (LOG.isTraceEnabled()) {
```
[the remark this answers is not in the document]
Comment (Secchol) [reply src/main/java/com/ontotext/kafka/service/GraphDBService.java:91]: This operation involves retrieving the user specifically to log the current username, whereas in the code above the logging utilizes properties that were already retrieved and are also used elsewhere in the code.

File: src/main/java/com/ontotext/kafka/service/UpdateRecordsProcessor.java (hunk 3 above)
Comment (Secchol) [reply src/main/java/com/ontotext/kafka/service/UpdateRecordsProcessor.java:54]: Having in mind the above comments, I've decided to remove these logs entirely

File: src/main/java/com/ontotext/kafka/util/PropertiesUtil.java (hunk 4 above)
Comment (Secchol) [reply src/main/java/com/ontotext/kafka/util/PropertiesUtil.java:27]: Nope, I missed logging the exception by accident.

**Review by ivorusev — APPROVED** [review]

**Pull request merged**